src/h3/mod.rs

```diff
@@ -608,6 +625,12 @@ impl Connection {
     pub fn send_request(
         &mut self, conn: &mut super::Connection, headers: &[Header], fin: bool,
     ) -> Result<u64> {
+        // If we received a GOAWAY from the peer, MUST NOT initiate new
```

Comment:
In draft 28+, GOAWAY is available to client and server but kinda asymmetric.

> Endpoints MUST NOT initiate new requests or promise new pushes on the
   connection after receipt of a GOAWAY frame from the peer.

The requirement are also looser for the server, if a client sends GOAWAY, it will reject pushes with Push IDs greater than or equal to this value (not detect a protocol error).

Reply:
- oh wait, I think that is just reflecting the race condition between client GOAWAY and push. Think it is a MUST NOT for the server to generate pushes too.